### ISSUE TITLE
Add tokenization support for bundle metadata when running porter build

### ISSUE THREAD

**Issue by MikeLarah** [getporter/porter#694]
It would be good to have a way to tokenize parts of the bundle metadata file (porter.yaml) when running `porter build`.

For example, we have the following scenarios:

1. We want to set the `version` of the bundle at build time, which would also change the `invocationImage` and `tag` properties.
2. We want to specify the registry for bundle at build time, which would affect the `invocationImage` and `tag` properties. (e.g. to support users who have forked a repo and want to push bundles into their own registry, to support multiple environments).

This would need parameter support in the `build` command (possibly using a similar mechanism used for `porter install`?)

**Comment by jeremyrickard** [comment]
@MikeLarah thanks for the issue! We have definitely been thinking about this!

**Comment by vdice** [comment]
Hi @MikeLarah , wanted to check in on this.  After updating metadata fields on the Porter manifest relevant to this issue in https://github.com/getporter/porter/issues/1333, @carolynvs and I chatted about desired build flag options (currently tracked in https://github.com/getporter/porter/issues/1334).

Looking at the two requests above, does the proposed functionality mentioned work for your needs?

> We want to set the version of the bundle at build time, which would also change the invocationImage and tag properties.

Fairly straightforward here: we indeed plan to allow users to supply a `version` override on `porter build`.  (Also, `name`)

> We want to specify the registry for bundle at build time, which would affect the invocationImage and tag properties. (e.g. to support users who have forked a repo and want to push bundles into their own registry, to support multiple environments).

We're currently leaning towards only allowing registry and reference overrides at the time of publish (e.g. `porter publish --registry myregistry`).  See https://github.com/getporter/porter/issues/1335.  Does this still support the use cases you had in mind?
